interpreter/prune.go

```diff
@@ -281,13 +281,35 @@ func (p *astPruner) prune(node ast.Expr) (ast.Expr, bool) {
 	}
 	if macro, found := p.macroCalls[node.ID()]; found {
 		// Ensure that intermediate values for the comprehension are cleared during pruning
+		pruneMacroCall := node.Kind() != ast.UnspecifiedExprKind
 		if node.Kind() == ast.ComprehensionKind {
 			compre := node.AsComprehension()
 			visit(macro, clearIterVarVisitor(compre.IterVar(), p.state))
+			if compre.HasIterVar2() {
+				visit(macro, clearIterVarVisitor(compre.IterVar2(), p.state))
+			}
+			loopCond := compre.LoopCondition()
+			iterRange := compre.IterRange()
+			pruneMacroCall =
+				(loopCond.Kind() == ast.LiteralKind && loopCond.AsLiteral() == types.False) &&
```

Comment:
nit/clarity -- could you add a comment or add a predicate function for this test (checking for trivial comprehension)? does this also apply to .optMap?

Reply:
Perhaps I should just restrict the pruning to `cel.bind` calls only since `optMap` although not wrong would look very strange. Thoughts?

Reply:
Just testing for cel.bind is easier to follow imo, but up to you.

Reply:
Done, PTAL.